packages/gatsby-plugin-sharp/src/utils.js

```diff
@@ -146,29 +164,30 @@ export const calculateImageSizes = ({
 
     // Create sizes (in width) for the image if no custom breakpoints are
     // provided. If the max width of the container for the rendered markdown file
-    // is 800px, the sizes would then be: 200, 400, 800, 1200, 1600.
+    // is 800px, the sizes would then be: 200, 400, 800, 1600, 2400 (based off of
+    // the default outputPixelDensities)
     //
     // This is enough sizes to provide close to the optimal image size for every
     // device size / screen resolution while (hopefully) not requiring too much
     // image processing time (Sharp has optimizations thankfully for creating
     // multiple sizes of the same input file)
 
-    // use standard breakpoints if no custom breakpoints are specified
-    if (!srcSetBreakpoints || !srcSetBreakpoints.length) {
-      sizes = densities
-        .map(density => density * width)
-        .filter(size => size <= maxWidth)
-      if (!sizes.includes(maxWidth)) {
-        sizes.push(maxWidth)
-      }
-      sizes = sizes.sort()
-    } else {
-      let sizes = srcSetBreakpoints.filter(size => size <= maxWidth)
-      if (!sizes.includes(maxWidth)) {
-        sizes.push(maxWidth)
-      }
-      sizes = sizes.sort()
+    sizes = densities.map(density => density * fixedValue)
+    // add breakpoint set to sizes if they are specified
+    if (srcSetBreakpoints) {
+      srcSetBreakpoints.forEach(breakpoint => sizes.push(breakpoint))
     }
+    sizes = sizes.filter(size => size <= imgDimensions.width)
+
+    // ensure that the size passed in is included in the final output
+    if (!sizes.includes(maxWidth)) {
+      sizes.push(maxWidth)
+    }
+    sizes = sizes.sort((a, b) => a - b)
```

Comment:
This is the default comparator, so you can just do `sizes.sort()`

Reply:
So turns out this has this fun behavior of sorting by character which is almost right 😅 :

```
[0, 10, 190, 120, 500, 5, 1000000].sort()
> [0, 10, 1000000, 120, 190, 5, 500]
```

vs.

```
[0, 10, 190, 120, 500, 5, 1000000].sort((a, b) => a - b)
> [0, 5, 10, 120, 190, 500, 1000000]
```

Reply:
Omg. TIL!